src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c

```diff
@@ -28,45 +37,543 @@ const ActorInit Demo_Syoten_InitVars = {
+            this->actionFunc = func_80C16EAC;
+            Actor_SetScale(&this->actor, 0.5f);
+            break;
+
+        case DEMOSYOTEN_F_4:
+            this->unk_3DC = object_syoten_DL_002880;
+            this->unk_3E0 = Lib_SegmentedToVirtual(&object_syoten_Matanimheader_002B88);
+            this->unk_3E4 |= 2;
+            this->actionFunc = func_80C17008;
+            this->unk_3F0 = 0x217;
+            this->unk_3E4 |= 1;
+            this->unk_3E4 |= 8;
+            this->actor.draw = func_80C17690;
+            Actor_SetScale(&this->actor, 5.0f);
+            break;
+    }
+}
+
+void DemoSyoten_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DemoSyoten* this = THIS;
+
+    if (DEMOSYOTEN_GET_F(&this->actor) == DEMOSYOTEN_F_0) {
+        func_8018349C((void*)this->unk_144);
+    }
+}
+
+void func_80C16760(DemoSyoten* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3s* points;
+    Vec3f sp2C;
+
+    this->unk_3EC = 0;
+    if (DEMOSYOTEN_GET_7E00(&this->actor) != DEMOSYOTEN_7E00_3F) {
+        this->unk_3E8 = &globalCtx->setupPathList[DEMOSYOTEN_GET_7E00(&this->actor)];
+        if (this->unk_3E8 != NULL) {
+            points = Lib_SegmentedToVirtual(this->unk_3E8->points);
+            Math_Vec3s_ToVec3f(&this->actor.world.pos, &points[0]);
+            this->unk_3EC++;
+            points++;
+            Math_Vec3s_ToVec3f(&sp2C, points);
+            this->actor.world.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &sp2C);
+            this->actor.world.rot.x = Math_Vec3f_Pitch(&this->actor.world.pos, &sp2C);
+        }
+    } else {
+        this->unk_3E8 = NULL;
+    }
+}
+
+s32 func_80C16818(DemoSyoten* this) {
+    s32 pad;
+    Path* path = this->unk_3E8;
+    Vec3s* points;
+    Vec3f sp28;
+
+    if (path == NULL) {
+        return true;
+    }
+
+    points = Lib_SegmentedToVirtual(this->unk_3E8->points);
+    points += this->unk_3EC;
+    Math_Vec3s_ToVec3f(&sp28, points);
+    this->actor.world.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &sp28);
+    this->actor.world.rot.x = Math_Vec3f_Pitch(&this->actor.world.pos, &sp28);
+    Actor_MoveWithoutGravityReverse(&this->actor);
+    if (Math_Vec3f_DistXYZ(&this->actor.world.pos, &sp28) < 2.0f) {
+        return true;
+    }
+    return false;
+}
+
+void func_80C168D0(DemoSyoten* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Path* path = this->unk_3E8;
+    Vec3s* points;
+    Vec3f sp30;
+    Vec3f sp24;
+    f32 sp20;
+
+    if (path != NULL) {
+        points = Lib_SegmentedToVirtual(this->unk_3E8->points);
+        points += this->unk_3EC;
+        Math_Vec3s_ToVec3f(&sp30, points);
+        Actor_GetProjectedPos(globalCtx, &sp30, &sp24, &sp20);
+        Math_Vec3f_Copy(&sp30, &this->actor.projectedPos);
+        sp30.z = sp30.y;
+        sp24.z = sp24.y;
+        this->actor.shape.rot.z = -Math_Vec3f_Yaw(&sp24, &sp30);
+    }
+}
+
+typedef struct {
+    /* 0x00 */ char unk00[0xF];
+    /* 0x0F */ s8 unk_0F;
+} DemoSyotenStruct; // size = 0x10
```

Comment:
Isn't this just a `Vtx`?

Reply:
Maybe, no idea.

Reply:
```c
DemoSyotenStruct* temp_v0 = Lib_SegmentedToVirtual(&object_syotenVtx_0018C0);
```
Given that this it's found by virtualising the pointer to a Vtx...

Reply:
Oh, yeah, this was a thing I didn't see until I had done everything, and post-hoc fixed it at the end. I fix these syms at the end, and 0x18C0 wasn't in the object xml and so I started making a blob for it, but then thought I'd check the object anyway and then I saw it was in there as a Vtx, so I just replaced the symbol name and forgot the code. I'll change it ty ty.